arangod/Aql/AqlItemBlockInputRange.cpp

```diff
@@ -57,6 +57,7 @@ bool AqlItemBlockInputRange::hasDataRow() const noexcept {
   return isIndexValid(_rowIndex) && !isShadowRowAtIndex(_rowIndex);
 }
 
+// TODO: Implement peekDataRow (without state). e.g. IResearchViewExecutor does not need the state!
```

Comment:
I agree on this todo, and I am happy to have it solved as soon as all Executors are merged